compose/project.py

```diff
@@ -473,6 +477,23 @@ def get_networks(service_dict, network_definitions):
     return networks
 
 
+def match_named_volumes(service_dict, project_volumes):
+    for volume_spec in service_dict.get('volumes', []):
+        if volume_spec.is_named_volume:
+            declared_volume = next(
+                (v for v in project_volumes if v.name == volume_spec.external),
+                None
+            )
+            if not declared_volume:
+                raise ConfigurationError(
+                    'Named volume "{0}" is used in service "{1}" but no'
+                    ' declaration was found in the volumes section.'.format(
+                        volume_spec.repr(), service_dict.get('name')
+                    )
+                )
```

Comment:
I think we should do this validation as part of `compose/config/config` similar to how we validate depends_on. That way we the `docker-compose config` command will catch the error (since that command is supposed to be able to validate a config).

Once we've validated that all the named volumes are defined, we can just add the prefix in service when we build the binds without worrying about looking it up.